rules/Transform/Rector/Class_/AddAllowDynamicPropertiesAttributeRector.php

```diff
@@ -139,30 +139,48 @@ private function addAllowDynamicPropertiesAttribute(Class_ $class): Class_
 
     private function shouldSkip(Class_ $class): bool
     {
-        $className = (string) $this->nodeNameResolver->getName($class);
-        foreach ($this->transformOnNamespaces as $transformOnNamespace) {
-            if (str_contains($transformOnNamespace, '*')) {
-                if (! fnmatch($transformOnNamespace, $className, FNM_NOESCAPE)) {
-                    return true;
-                }
+        if ($this->isDescendantOfStdclass($class)) {
+            return true;
+        }
 
-                continue;
-            }
+        if ($this->hasNeededAttributeAlready($class)) {
+            return true;
+        }
+
+        if ($this->hasMagicSetMethod($class)) {
+            return true;
+        }
 
-            if ($this->nodeNameResolver->isStringName($className, $transformOnNamespace)) {
+        $className = $this->getName($class);
```

Comment:
unrelated idea I got from looking at this example: maybe we should have a conditional return-type on `getName()` which makes it return a `class-string` when the arg passed is a `Class_`